docs/docs/code/python/README.md

```diff
@@ -0,0 +1,279 @@
+
+<div>
+<img alt="Python print log output in the results" src="./images/print-logs.png">
+</div>
+
+## Using 3rd party packages
+
+You can use any packages from [PyPi](https://pypi.org) in your Pipedream workflows. This includes popular choices such as:
+
+* `requests` for making HTTP requests
+* `sqlalchemy` for retrieving or inserting data in a SQL database
+* `pandas` for working with complex datasets
+
+To use a PyPi package, just include it in your step's code:
+
+`import requests`
+
+And that's it.
+
+No need to update a `requirements.txt` or specify elsewhere in your workflow of which packages you need. Pipedream will automatically install the dependency for you.
+
+## Making an HTTP request
+
+We recommend using the popular `requests` HTTP client package available in Python to send HTTP requests.
+
+No need to run `pip install`, just `import requests` at the top of your step's code and it's available for your code to use.
+
+### Making a GET request
+
+GET requests typically are for retrieving data from an API. Below is an example d
+
+```python
+import requests
+
+url = 'https://swapi.dev/api/people/1'
+
+r = requests.get(url)
+
+# to see the response in your Pipedream step logs:
+print(r.text)
+
+# to see the response status code in your Pipedream step logs:
+print(r.status)
+```
+
+### Making a POST request
+
+```python
+import requests
+
+# This a POST request to this URL will echo back whatever data we send to it
+url = 'https://postman-echo.com/post'
+
+data = {"name": "Bulbasaur"}
+
+r = requests.post(url, data)
+
+# to see the response in your Pipedream step logs:
+print(r.text)
+
+# to see the response status code in your Pipedream step logs:
+print(r.status)
+```
+
+### Using API key authentication
+
+If an particular service requires you to use an API key, you can pass it via the headers of the request.
+
+This proves your identity to the service so you can interact with it:
+
+```python
+import requests
+
+token = 'replace this with your API key'
+
+url = 'https://api.twitter.com/2/users/@pipedream/mentions'
+
+headers { 'Authorization': f"Bearer {token}"}
+r = requests.get(url, headers=headers)
+
+print(r.text)
+```
+
+```note
+Not all APIs use a `Authorization: Bearer <api key>` header format.
+
+Refer to the documentation of the API you're working with for their specific authentication format.
+```
+
+### Sending files
+
+You can send also files within a step.
+
+An example of sending a previously stored file in the workflow's `/tmp` directory: 
+
+```python
+# Retrieving a previously saved file from workflow storage
+files = {'image': ('python-logo.png', open('/tmp/python-logo.png', 'rb')}
+
+r = requests.post(url='https://api.imgur.com/3/image', files=files)
+```
+
+## Sharing data between steps
+
+You can share data between steps in your workflows. Data can be passed forward to the next steps, and it can be used by other steps downstream.
```

Comment:
Like above, I would mimic the language in the Bash section to keep each of these introductions consistent.